backend/api/Controllers/MissionController.cs

```diff
@@ -112,23 +118,53 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string id)
         [FromBody] ScheduledMissionQuery scheduledMissionQuery
     )
     {
+        var robot = await _robotService.ReadById(scheduledMissionQuery.RobotId);
+        if (robot is null)
+            return NotFound($"Could not find robot with id {scheduledMissionQuery.RobotId}");
+
+        EchoMission? echoMission;
         try
         {
-            var newMission = await _missionService.Create(scheduledMissionQuery);
-            return CreatedAtAction(nameof(GetMissionById), new { id = newMission.Id }, newMission);
-        }
-        catch (KeyNotFoundException e)
-        {
-            return NotFound(e.Message);
+            echoMission = await _echoService.GetMissionById(scheduledMissionQuery.EchoMissionId);
         }
         catch (HttpRequestException e)
         {
-            return StatusCode(StatusCodes.Status502BadGateway, e.Message);
+            if (e.StatusCode.HasValue && (int)e.StatusCode.Value == 404)
+            {
+                _logger.LogWarning(
+                    "Could not find echo mission with id={id}",
+                    scheduledMissionQuery.EchoMissionId
+                );
+                return NotFound("Echo mission not found");
+            }
+
+            _logger.LogError(e, "Error getting mission from Echo");
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
```

Comment:
Do we want to do the logging in the service? then the controller just catches exceptions and returns appropriate StatusCode?

Reply:
Why should we not log?

Reply:
We want to log, but I want to log in the service rather than in the controller. Because if several controllers use the same service we want the same log message if the service fail, and also remove some code from the controllers